wl/server/TransifexIntegration.java

```diff
@@ -0,0 +1,256 @@
+			for (List l : relevantIssues.values()) total += l.size();
+
+			File message = File.createTempFile("temp", null);
+			PrintWriter write = new PrintWriter(message);
+			write.println("From: noreply@widelands.org");
+			write.println("Subject: Transifex String Issues");
+			write.println();
+
+			write.print("Dear " + uploader + ",\nthe translators have found " + total +
+			            " new issue(s) in " + relevantIssues.size() +
+			            " of your add-on(s). Below you may find a list of all new string issues.");
+			for (String addon : relevantIssues.keySet()) {
+				List<Issue> list = relevantIssues.get(addon);
+				write.print(
+				    "\n\n################################################################################\n " +
+				    list.size() + " new issue(s) in add-on " + addon);
+				for (Issue i : list) {
+					write.print(
+					    "\n --------------------------------------------------------------------------------"
+					    + "\n  Issue ID      : " + i.issueID + "\n  Source String : " + i.string +
+					    "\n  String ID     : " + i.stringID +
+					    "\n  Occurrences   : " + i.occurrence + "\n  Issue message : " + i.message);
+				}
+				write.print(
+				    "\n################################################################################");
+			}
+
+			write.print(
+			    "\n\n-------------------------\n"
+			    +
+			    "To change how you receive notifications, please go to https://www.widelands.org/notification/.");
+			write.close();
+			Utils.bash("bash", "-c", "ssmtp '" + email + "' < " + message.getAbsolutePath());
+			message.delete();
+		}
+	}
+
+	private synchronized List<Issue> fetchIssues() throws Exception {
+		ContainerFactory cf = new ContainerFactory() {
+			public List creatArrayContainer() { return new LinkedList(); }
+			public Map createObjectContainer() { return new LinkedHashMap(); }
+		};
+
+		JSONParser parser = new JSONParser();
+		Map json = (Map)parser.parse(
+		    Utils.bashOutput("curl", "-g", "-H",
+		                     "Authorization: Bearer " + Utils.config("transifextoken"),
+		                     "https://rest.api.transifex.com/"
+		                         + "resource_string_comments?"
+		                         + "filter[organization]=o:widelands&"
+		                         + "filter[type]=issue&"
+		                         + "filter[project]=o:widelands:p:widelands-addons&"
+		                         + "filter[status]=open"),
+		    cf);
+
+		List<Issue> result = new ArrayList<>();
+		for (Object oneIssue : (List)json.get("data")) {
+			String sourceStringURL =
+			    ((Map)((Map)((Map)((Map)oneIssue).get("relationships")).get("resource_string"))
+			         .get("links"))
+			        .get("related")
+			        .toString();
+			Map sourceStringQuery = (Map)parser.parse(
+			    Utils.bashOutput("curl", "-g", "-H",
+			                     "Authorization: Bearer " + Utils.config("transifextoken"),
+			                     sourceStringURL),
+			    cf);
```

Comment:
Is there an API limit on transifex? If yes, a query for each issue may reach this limit (no idea, how many issues there are...)

Reply:
We have currently 12 open issues. The most I've ever seen were 25 or so. So I doubt we will ever hit the limit:
> To maintain optimal performance for our API and ensure its available to all our customers, we limit the request rate to 30,000 requests/hour or 5,000 requests / 5 minutes / IP. Once past that limit, requests will be throttled and you will get a 429 error code.

In practice we might get into trouble much earlier than that: when the number of issues exceeds 150, due to pagination. But I'll deal with that when that ever happens :D